XenAdmin/Controls/NetworkingTab/NetworkList.cs

```diff
@@ -509,7 +509,7 @@ private void AddNetworkButton_Click(object sender, EventArgs e)
                     // Cache populating?
                     return;
                 }
-                VIFDialog d = new VIFDialog(vm.Connection, null, VIF.GetDeviceId(vm));

```

Comment:
Instead of passing the VM object to the dialog, we could just pass a bool (e.g. allowSriov) that is calculated here.